overlord/snapstate/snapmgr.go

```diff
@@ -439,28 +434,40 @@ func (m *SnapManager) ensureRefreshes() error {
 
 	tr := config.NewTransaction(m.state)
 
-	// allow disabling auto-refresh in the tests
-	if osutil.GetenvBool("SNAPD_DEBUG") {
-		var refreshDisabled bool
-		err := tr.Get("core", "refresh.disabled", &refreshDisabled)
-		if err != nil && !config.IsNoOption(err) {
-			return err
-		}
-		if refreshDisabled {
-			return nil
-		}
+	// get last refresh time
+	var lastRefresh time.Time
+	err := m.state.Get("last-refresh", &lastRefresh)
+	if err != nil && err != state.ErrNoState {
+		return err
 	}
 
-	var lastRefresh time.Time
-	err := tr.Get("core", "refresh.last", &lastRefresh)
+	// get our schedule
+	refreshScheduleStr := defaultRefreshSchedule
+	err = tr.Get("core", "refresh.schedule", &refreshScheduleStr)
 	if err != nil && !config.IsNoOption(err) {
 		return err
 	}
-
-	nextRefresh := lastRefresh.Add(minRefreshInterval).Add(m.refreshRandomness)
-	if time.Now().Before(nextRefresh) {
-		return nil
+	refreshSchedule, err := timeutil.ParseSchedule(refreshScheduleStr)
+	if err != nil {
+		// FIXME: this will spam syslog :/
+		logger.Noticef("cannot use refresh.schedule: %s", err)
+		refreshSchedule, err = timeutil.ParseSchedule(defaultRefreshSchedule)
+		if err != nil {
+			panic(fmt.Sprintf("defaultRefreshSchedule cannot be parsed: %s", err))
+		}
 	}
+	// already have a refresh timer
+	if m.nextRefresh != nil {
+		if m.currentRefreshSchedule == refreshScheduleStr {
+			return nil
+		}
+		// the refresh schedule has changed
+		logger.Debugf("Refresh-schedule changed, reloading")
```

Comment:
We should try to respect the casing of settings, so perhaps one of:
 - `Option refresh-schedule changed, reloading.`
 - `Refresh schedule changed, reloading.`